File: primes/perl5/primes.pl
#!/usr/bin/perl
use strict;

if (2 != @ARGV) {
	die("usage: $0 <start> <n>\n");
}

my $num = shift @ARGV;  # number to start from
my $max = shift @ARGV;  # number of primes to find

my $count = 0;
while ($count < $max) {

	my $found = 1;  # default, found=true
	for (my $i = 2; $i < $num; $i++) {
		if (0 == ($num % $i)) {
			$found = 0;
			last;
		}
	}

	if ($found) {
		print "$num\n";
		$count++;
	}

	$num++;
}

File: primes/octave/primes.m
#!/usr/bin/octave -qf

if (2 != length(argv()))
	printf('usage: %s <start> <n>', program_name());
	exit(1);
endif

num = str2num(argv(){1});
max = str2num(argv(){2});

count = 0;
while count < max
	found = 1;
	for i = 2:(num - 1)
		if (0 == mod(num, i))
			found = 0;
			break;
		endif
	endfor

	if found
		printf('%i\n', num)
		count = count + 1;
	endif

	num = num + 1;
endwhile
